api/app/tests/apps/email_integrations/test_utils.py

```diff
@@ -0,0 +1,172 @@
+# SPDX-License-Identifier: MPL-2.0
+# Copyright (C) 2021 Gemeente Amsterdam
+import re
+
+from django.conf import settings
+from django.test import TestCase
+from faker import Faker
+
+from signals.apps.email_integrations.factories import EmailTemplateFactory
+from signals.apps.email_integrations.models import EmailTemplate
+from signals.apps.email_integrations.utils import (
+    URL_PATTERN,
+    make_email_context,
+    validate_email_template,
+    validate_template
+)
+from signals.apps.signals.factories import SignalFactory
+
+
+class TestUtils(TestCase):
+    def test_regex(self):
+        """
+        Check if a URL matches the URL_PATTERN
+        Check if the URL is replaced by ''
+        Check if the URL is replaced by '' if we put it in a random text
+
+        We test with a couple of examples we can think of and a couple of examples created by the Faker
+        """
+        test_schemas = ['http://', 'https://', 'ftp://', 'sftp://', ]
+        test_uris = [
+            'test-domain.com',
+            'www.test-domain.com/',
+            'test-domain.com?query=param',
+            'test-domain.com/?query=param',
+            'www.test-domain.com/with/path?query=param',
+            'test-domain.com/with/path/?query=param',
+            'test-domain.com?query=param&extra=param',
+            'test-domain.com/?query=param&extra=param',
+            'www.test-domain.com/with/path?query=param&extra=param',
+            'test-domain.com/with/path/?query=param&extra=param',
+            'test-domain.com:8080',
+            'www.test-domain.com:8080/',
+            'user:password@test-domain.com',
+            'user:password@test-domain.com/',
+            'user:password@test-domain.com:8080',
+            'user:password@www.test-domain.com:8080/',
+            'test-domain.com:8080?query=param',
+            'test-domain.com:8080/?query=param&extra=param',
+            'user:password@test-domain.com?query=param',
+            'user:password@test-domain.com/?query=param&extra=param',
+            'user:password@test-domain.com:8080?query=param',
+            'user:password@test-domain.com:8080/?query=param&extra=param',
+            'test-domain.co.uk',
+            'www.test-domain.co.uk/',
+            'úêï.google',
+            'www.úêï.google/',
+            'strangetld.lol',
+            'strangetld.fun',
+            'strangetld.wow',
+            'strangetld.unicorn',
+            'www.example.com/just/a/path/',
+            'test.com',
+        ]
+
+        fake = Faker()
+        fake_text = fake.text()
+        position_to_insert = fake_text.find('.')
```

Comment:
This assumes the text will contain a period, is that always true? Consider `random.randrange` to grab a random valid index in the generated text --- otherwise tests may fail randomly.

Reply:
It creates a couple of sentences with a period. Documentation: https://faker.readthedocs.io/en/stable/providers/faker.providers.lorem.html#faker.providers.lorem.Provider.text